src/sparsezoo/objects/directories.py

```diff
@@ -283,14 +284,90 @@ class OnnxGz(Directory):
+        """
+        Assumptions:
+            - the tarball contains only one onnx model, with the
+                name `model.onnx`
+            - the tarball will be extracted to the same directory as the tarball
+
+        :post-condition: self._path will point to the path of the extracted
+            onnx model, and it exists
+        :return: path to the onnx model
+        """
+
+        expected_path: Path = (
+            Path(self._path)
+            if self._path is not None
+            else Path(self.parent_directory) / "model.onnx.tar.gz"
+        )
+
+        # _path can also point to parent directory of the tarball when
+        #  this object is initialized from files
+        if expected_path.is_dir():
+            # point expected to model.onnx.tar.gz
+            expected_path = expected_path / "model.onnx.tar.gz"
+
+        # point path to model.onnx.tar.gz before download/unzip
+        self._path = str(expected_path)
+
+        if not expected_path.exists():
+            # download the tarball if it does not exist
+            self.download()
+
+        self._check_if_extracted()
         if self.is_archive:
+            # if the tarball is not extracted, extract it
             self.unzip()
-        if os.path.isdir(self._path) and "model.onnx" in os.listdir(self._path):
-            # if unzipped into a directory, refer directly to model.onnx
-            self._path = os.path.join(self._path, "model.onnx")
+
+        onnx_model_path = expected_path.with_name(name="model.onnx")
+        if not onnx_model_path.exists():
+            raise FileNotFoundError(
+                f"Expected to find model.onnx at {onnx_model_path}, "
+                "but it does not exist."
+            )
+
+        # point _path to model.onnx
+        self._path = str(onnx_model_path)
         return self._path
+
+    def _check_if_extracted(self):
+        """
+        set `is_archive` to False if the tarball is extracted, else set it to
+        True. Condition for being extracted is that all members of the tarball
+        are extracted and exist as files in the same directory as the tarball.
+        """
+        # expected file will point to the tarball
+        expected_path: Path = (
+            Path(self._path)
+            if self._path is not None
+            else Path(self.parent_directory) / "model.onnx.tar.gz"
+        )
+
+        if expected_path.is_dir():
+            # point expected to model.onnx.tar.gz
+            expected_path = expected_path / "model.onnx.tar.gz"
+
+        model_gz_path = expected_path.with_name(name="model.onnx.tar.gz")
+        # assert all members of  model.onnx.tar.gz have been extracted
+        for zipped_filename in tarfile.open(model_gz_path).getnames():
+            unzipped_file_path = expected_path.with_name(zipped_filename)
+            if not unzipped_file_path.exists():
+                _LOGGER.info(f"{unzipped_file_path} does not exist, was it extracted?")
```

Comment:
should remove log - user can't do anything with this. potentially move to debug since this would be a weird state